src/sentry/tasks/sentry_apps.py

```diff
@@ -386,6 +381,9 @@ def send_and_save_webhook_request(sentry_app, app_platform_event, url=None):
         elif resp.status_code == 504:
             raise ApiTimeoutError.from_request(resp.request)
 
+        if 400 <= resp.status_code <= 404:
```

Comment:
@MeredithAnya I think we should check up to status code `499` since there are other types of client errors (even if they are less common).